### PR TITLE
Adds Kind Host-to-Container Port Mapping

### PR DESCRIPTION
**Pull request by danehans**
Maps host port 9080 to container port 80 and host port 9443 to container port 443 for kind cluster.

Fixes: #284 

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

**Comment by danehans** [comment]
@skriss PTAL when you have a moment. We need this for RC1.

File: tools/hack/kind-expose-ports.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+  - role: worker
+    extraPortMappings:
+      - containerPort: 80
+        hostPort: 9080
```
[the remark this answers is not in the document]
Comment (skriss) [reply tools/hack/kind-expose-ports.yaml:8]: The conformance tests send requests to whatever address is in the Gateway's status.addresses field. It needs to be routable from wherever the conformance tests are run.

xref. #330

File: tools/hack/create-cluster.sh [hunk 2]
```diff
@@ -6,12 +6,16 @@ set -euo pipefail
 CLUSTER_NAME=${CLUSTER_NAME:-"envoy-gateway"}
 METALLB_VERSION=${METALLB_VERSION:-"v0.12.1"}
 KIND_NODE_TAG=${KIND_NODE_TAG:-"v1.24.0"}
+readonly HERE=$(cd "$(dirname "$0")" && pwd)
+readonly REPO=$(cd "${HERE}/../.." && pwd)
+readonly CONFIG_FILE="${REPO}/tools/hack/kind-expose-ports.yaml"
```
Comment (LukeShu) [inline tools/hack/create-cluster.sh:11]: Ugh, I hate that these types of hacks are necessary.  We solved this long ago with `realpath`, but macOS is holding the world back...

But wait, we have `git`!
```bash
REPO=$(git rev-parse --show-toplevel)
```

File: tools/hack/kind-expose-ports.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
```
Comment (LukeShu) [inline tools/hack/kind-expose-ports.yaml:4]: Please rebase/merge with `main`; the yamllint config is going to complain about this whitespace.

**Review by LukeShu — CHANGES_REQUESTED** [review]

**Comment by danehans** [comment]
I am closing this issue since this functionality is not required for conformance tests and metallb does not work with Mac, [xref](https://kind.sigs.k8s.io/docs/user/loadbalancer/).

**Pull request closed without merging**